datumaro/components/operations.py

```
@@ -145,14 +149,43 @@ def _merge_attrs(a: Dict[str, Any], b: Dict[str, Any],
 
         return merged
 
+    @classmethod
+    def _merge_media(cls, item_a: DatasetItem, item_b: DatasetItem) \
+            -> Union[ByteImage, Image, PointCloud, Video, VideoFrame]:
+        if isinstance(item_a.media, Image) or isinstance(item_b.media, Image):
+            media = cls._merge_images(item_a, item_b)
+        elif isinstance(item_a.media, PointCloud) and isinstance(item_b.media, PointCloud):
+            media = cls._merge_point_clouds(item_a, item_b)
+        elif isinstance(item_a.media, Video) and isinstance(item_b.media, Video):
+            media = cls._merge_videos(item_a, item_b)
+        else:
+            if isinstance(item_a.media, MediaElement) and isinstance(item_b.media, MediaElement):
+                if item_a.media.path and item_b.media.path and \
+                        item_a.media.path != item_b.media.path:
+                    raise MismatchingMediaPathError(
+                        (item_a.id, item_a.subset),
+                        item_a.media.path, item_b.media.path)
+
+                if item_a.media.path:
+                    media = item_a.media
+                else:
+                    media = item_b.media
+
+            elif isinstance(item_a.media, MediaElement):
+                media = item_a.media
+            else:
+                media = item_b.media
```

This block is only valid if one of items doesn't have media info. If both items have media, but the types are different, we definitely need to raise an error.

---

Added an error for this